### PR TITLE
Add test for recursive_conservative copy

### PR DESCRIPTION
**Pull request by kersommoura**
Add test for recursive_conservative copy. Explore the newly added flag
`recursive_conservative` used together with `recursive`.

See: https://pulp.plan.io/issues/4269

### CODE REVIEW

**Comment by kersommoura** [comment]
@dralley @goosemania, please, confirm that those test scenarios described are sane.

**Comment by kersommoura** [comment]
@dralley, I do have tests for simple copy. But I will add this scenario that you described, since we can now use the ``recursive_conservative`` as part of the simple copy.

File: pulp_2_tests/tests/rpm/api_v2/test_copy.py [hunk 1]
```diff
@@ -230,3 +232,198 @@ def test_all(self):
...
+    RPM packages used in this test case::
+
+        chimpanzee
+        ├── squirrel
+        │   ├── camel
+        │   └── fox
+        └── walrus
+
+    chimpanzee has as dependencies: squirrel and walrus RPM packages.
+    squirrel has as dependencies: camel and fox RPM packages.
```
Comment (bherrin3) [inline pulp_2_tests/tests/rpm/api_v2/test_copy.py]:
```suggestion
    squirrel has dependencies: camel and fox RPM packages.
```
Reading `has as` just struck me weird.

File: pulp_2_tests/tests/rpm/api_v2/test_copy.py [hunk 2]
```diff
@@ -230,3 +232,198 @@ def test_all(self):
...
+
+    RPM packages used in this test case::
+
+        chimpanzee
+        ├── squirrel
+        │   ├── camel
+        │   └── fox
+        └── walrus
+
+    chimpanzee has as dependencies: squirrel and walrus RPM packages.
```
Comment (bherrin3) [inline pulp_2_tests/tests/rpm/api_v2/test_copy.py]:
```suggestion
    chimpanzee has dependencies: squirrel and walrus RPM packages.
```
or just change to 

```suggestion
    chimpanzee has dependencies on RPM packages squirrel and walrus.
```

File: pulp_2_tests/tests/rpm/api_v2/test_copy.py [hunk 3]
```diff
@@ -230,3 +232,198 @@ def test_all(self):
...
+        self.assertEqual(versions[0], '0.71', versions)
+
+        dst_unit_ids = [
+            unit['metadata']['name']
+            for unit in search_units(self.cfg, repo, {'type_ids': ['rpm']})
+        ]
+        self.assertEqual(len(dst_unit_ids), 5, dst_unit_ids)
+
+    def test_norecursive_conservative_dependency(self):
+        """Non-recursive, conservatve, with old dependency.
```
Comment (bherrin3) [inline pulp_2_tests/tests/rpm/api_v2/test_copy.py]:
```suggestion
        """Non-recursive, conservative, with old dependency.
```

File: pulp_2_tests/tests/rpm/api_v2/test_copy.py [hunk 4]
```diff
@@ -230,3 +232,198 @@ def test_all(self):
...
+
+        Do the following:
+
+        1. Copy ``chimpanzee`` RPM package from repository A to B using:
+           ``recursive`` as False, ``recursive_conservative`` as True, and
+           an older version of walrus paackage is present on the repo B
```
Comment (bherrin3) [inline pulp_2_tests/tests/rpm/api_v2/test_copy.py]:
```suggestion
           an older version of walrus package is present on the repo B
```

**Pull request merged**